tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/NonParticipatingTckResource.java

```diff
@@ -82,7 +84,7 @@ public Response neverRunWithLRA(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
 
     @PUT
     @Path(SUPPORTS_PATH)
-    @LRA(value = LRA.Type.SUPPORTS)
+    @LRA(value = LRA.Type.SUPPORTS, end = false)
```

Comment:
Will this change result in any active LRAs when the test finishes. For example, will the LRA in the following call ever be closed:
 
https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L232

Reply:
I checked with Narayana and there are no open LRAs after full TCK pass.

Reply:
@xstefank sorry for ignorance but why is needed to change the `SUPPORTS` here for not ending the LRA?

Reply:
hah, refreshing my memory :) So I think that the issue was that in some test the LRA was closed here and then propagated to some LRA resource into which I added the AfterLRA method (you can check by removing this change if you wish to know which test it was). So with AfterLRA the already ended LRA was rejected because only Cancelling/Closing LRAs are allowed with the AfterLRA method present.

Reply:
Where is the LRA closed or cancelled?

Reply:
So the affected test is [TckContextTests#testContextAfterRemoteCalls](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/TckContextTests.java#L255) which only calls [this JAX-RS method](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L214). The test calls the method which we are discussing in the [getActiveLRA()](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L214) method. As you can see in the [test flow](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L214), the test starts a new LRA and then on line [216](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L216) it calls this SUPPORTS_PATH resource method with this LRA where the LRA is currently closed (as it does have end = true which is the default). Then after some checks the `getActiveLRA()` is called again on line [224](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L224) where this SUPPORTS_PATH resource method is called again with already closed LRA.

As I added `@AfterLRA` method to `NonParticipatingTckResource` it is no longer truly non participating as it is an After LRA listener. So the invocation on line [224](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L224) fails as the LRA is already ended and AfterLRA listener requires only Active, Closing or Cancelling LRA.

Adding `end=false` fixes the issue and the LRA is closed at the end of [contextCheck](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L214) method.

Reply:
@mmusgrov ^^

Reply:
> Adding `end=false` fixes the issue and the LRA is closed at the end of [contextCheck](https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L214) method.

The end of this method is:

``` java
        // invoke a remote service which runs in the callers context, ie do set the context header
        remote = restPutInvocation(active, TCK_NON_PARTICIPANT_RESOURCE_PATH, SUPPORTS_PATH, "");
        assertEquals("contextCheck4: the remote service should have ran with the active context", remote, active);
        assertEquals("contextCheck4: after calling a remote service the active LRAs is different", active, getActiveLRA());

        return Response.ok().entity(lraId.toASCIIString()).build();
```

so how does it get closed?

Reply:
@mmusgrov the LRA is closed normally at the end of `contextCheck` method which has and LRA annotation -- `@LRA(value = LRA.Type.REQUIRED)`.

Reply:
I think I follow what @xstefank is saying about `contextCheck`.